src/Main/Java/com/hillel/vynnyk/homeworks/homework7/FileNavigator.java

```diff
@@ -0,0 +1,89 @@
+package com.hillel.vynnyk.homeworks.homework7;
+
+import java.util.*;
+
+
+public class FileNavigator {
+    Map<String, ArrayList<FileData>> map = new HashMap<>();
+
+    public Map<String, ArrayList<FileData>> add(String path, FileData file) {
+        ArrayList<FileData> data = new ArrayList<>();
+        if (map.containsKey(path)) {
+            map.get(path).add(file);
+        } else {
+            data.add(file);
+            map.put(path, data);
+        }
+        System.out.println(map);
+        return map;
+    }
+
+    public ArrayList<FileData> find(String key) {
+        if (map.containsKey(key)) {
+            return map.get(key);
+        }
+        return null;
+    }
+
+    public ArrayList<FileData> filterBySize(int size) {
+        ArrayList<FileData> data = new ArrayList<>();
+        for (ArrayList<FileData> fileData : map.values()) {
+            for (FileData file : fileData) {
+                if (file.getSize() <= size) {
+                    data.add(file);
+                }
+            }
+        }
+        return data;
+    }
+
+    public Map<String, ArrayList<FileData>> remove(String key) {
```

Comment:
не дуже гарно повертати внутрішній стан твого класу. Для remove можна повернути (опціонально) лише те що було видалено (тобто список для відповідного шляху)
```suggestion
    public void remove(String path) {
```